src/Components/Filters/filter.jsx

```diff
@@ -167,36 +277,78 @@ const Filter = () => {
-                name="punisher"
-                checked={punisher}
-                onChange={()=>{
-                  dispatch({type:Abbreviations.PU})
-                }}
-                
-                /> The
-                Punisher
+              <label htmlFor="punisher" className="label-selector">
+                <input
+                  type="checkbox"
+                  className="checkbox-select"
+                  id="punisher"
+                  name="punisher"
+                  checked={punisher}
+                  onChange={() => {
+                    dispatch({ type: Abbreviations.PUNISHER });
+                  }}
+                />{" "}
+                The Punisher
               </label>
             </li>
             <li>
-              <label for="filter-label" className="label-selector">
-                <input type="checkbox" className="checkbox-select" /> Daredevil
+              <label htmlFor="daredevil" className="label-selector">
+                <input
+                  type="checkbox"
+                  className="checkbox-select"
+                  id="daredevil"
+                  name="daredevil"
+                  checked={daredevil}
+                  onChange={() => {
+                    dispatch({ type: Abbreviations.DAREDEVIL });
+                  }}
+                />{" "}
+                Daredevil
               </label>
             </li>
             <li>
-              <label for="filter-label" className="label-selector">
-                <input type="checkbox" className="checkbox-select" /> Venom
+              <label htmlFor="venom" className="label-selector">
+                <input
+                  type="checkbox"
+                  className="checkbox-select"
+                  id="venom"
+                  name="venom"
+                  checked={venom}
+                  onChange={() => {
+                    dispatch({ type: Abbreviations.VENOM });
+                  }}
+                />{" "}
+                Venom
               </label>
             </li>
             <li>
-              <label for="filter-label" className="label-selector">
-                <input type="checkbox" className="checkbox-select" /> Deadpool
+              <label htmlFor="deadpool" className="label-selector">
+                <input
+                  type="checkbox"
+                  className="checkbox-select"
+                  id="deadpool"
+                  name="deadpool"
+                  checked={deadpool}
+                  onChange={() => {
+                    dispatch({ type: Abbreviations.DEADPOOL });
+                  }}
+                />{" "}
+                Deadpool
               </label>
             </li>
             <li>
-              <label for="filter-label" className="label-selector">
-                <input type="checkbox" className="checkbox-select" /> Loki
+              <label htmlFor="loki" className="label-selector">
```

Comment:
This file became too big. Make components for reused elements and use them to keep the code clean.